Firebase/Database/Api/FIRDatabaseComponent.m

```diff
@@ -75,33 +77,14 @@ + (void)load {
   return @[ databaseProvider ];
 }
 
-#pragma mar - Instance management.
-
-/**
- * A static NSMutableDictionary of FirebaseApp name and FRepoInfo to
- * FirebaseDatabase instance. To ensure thread-safety, it should only be
- * accessed in databaseForApp:URL:, which is synchronized.
- *
- * TODO: This serves a duplicate purpose as RepoManager.  We should clean up.
- * TODO: We should maybe be conscious of leaks and make this a weak map or
- * similar but we have a lot of work to do to allow FirebaseDatabase/Repo etc.
- * to be GC'd.
- */
-+ (FIRDatabaseDictionary *)instances {
-  static dispatch_once_t pred = 0;
-  static FIRDatabaseDictionary *instances;
-  dispatch_once(&pred, ^{
-    instances = [NSMutableDictionary dictionary];
-  });
-  return instances;
-}
+#pragma mark - Instance management.
 
 - (void)appWillBeDeleted:(FIRApp *)app {
   NSString *appName = app.name;
   if (appName == nil) {
     return;
   }
-  FIRDatabaseDictionary* instances = [FIRDatabaseComponent instances];
+  FIRDatabaseDictionary* instances = [self instances];
   @synchronized (instances) {
     NSMutableDictionary<FRepoInfo *, FIRDatabase *> *databaseInstances = instances[appName];
```

Comment:
This needs to   change based on the above change. Also, it doesn't look like we actually use FRepoInfo as a key here, so maybe it  makes sense to use something else (URL?)